runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -94,28 +95,44 @@ public void updateLayout(Layout layout, long rank)
      * @throws WrongEpochException wrong epoch number.
      */
     @SuppressWarnings("unchecked")
-    public Layout prepare(long epoch, long rank, Layout layout)
+    public Layout prepare(long epoch, long rank)
             throws QuorumUnreachableException, OutrankedException, WrongEpochException {
 
-        CompletableFuture<LayoutPrepareResponse>[] prepareList = layout.getLayoutClientStream()
-                .map(x -> x.prepare(epoch, rank))
+        CompletableFuture<LayoutPrepareResponse>[] prepareList = getLayout().getLayoutServers()
+                .stream()
```

Comment:
Since you are no longer using a copy of the layout here, can getLayout change after you set the layout manually in the previous step?

The client is used other places, and im worried other invocations can actually change the layout after it is set, as a consequence you can get the old one. Seems like a race?

Reply:
Yes, good catch. I'll store the current/old layout and use it in all the steps.

Reply:
Yes, good catch.